src/typescript-service.ts

```diff
@@ -991,6 +1038,106 @@ export class TypeScriptService {
 	}
 
 	/**
+	 * The code action request is sent from the client to the server to compute commands for a given
+	 * text document and range. These commands are typically code fixes to either fix problems or to
+	 * beautify/refactor code.
+	 *
+	 * @return Observable of JSON Patches that build a `Command[]` result
+	 */
+	textDocumentCodeAction(params: CodeActionParams, span = new Span()): Observable<OpPatch> {
+		const uri = normalizeUri(params.textDocument.uri);
+		return this.projectManager.ensureReferencedFiles(uri, undefined, undefined, span)
+			.toArray()
+			.mergeMap(() => {
+				const configuration = this.projectManager.getParentConfiguration(uri);
+				if (!configuration) {
+					throw new Error(`Could not find tsconfig for ${uri}`);
+				}
+				configuration.ensureBasicFiles(span);
+
+				const filePath = uri2path(uri);
+				const sourceFile = this._getSourceFile(configuration, filePath, span);
+				if (!sourceFile) {
+					throw new Error(`Expected source file ${filePath} to exist in configuration`);
+				}
+
+				const start = ts.getPositionOfLineAndCharacter(sourceFile, params.range.start.line, params.range.start.character);
+				const end = ts.getPositionOfLineAndCharacter(sourceFile, params.range.end.line, params.range.end.character);
+
+				const errorCodes = iterate(params.context.diagnostics)
+					.map(diagnostic => diagnostic.code)
+					.filter(code => typeof code === 'number')
+					.toArray() as number[];
+
+				return configuration.getService().getCodeFixesAtPosition(filePath, start, end, errorCodes, this.settings.format || {}) || [];
+			})
+			.map((action: ts.CodeAction): AddPatch => ({
+				op: 'add',
+				path: '/-',
+				value: {
+					title: action.description,
+					command: 'codeFix',
+					arguments: action.changes
+				} as Command
+			}))
+			.startWith({ op: 'add', path: '', value: [] } as AddPatch);
+	}
+
+	/**
+	 * The workspace/executeCommand request is sent from the client to the server to trigger command
+	 * execution on the server. In most cases the server creates a WorkspaceEdit structure and
+	 * applies the changes to the workspace using the request workspace/applyEdit which is sent from
+	 * the server to the client.
+	 */
+	workspaceExecuteCommand(params: ExecuteCommandParams, span = new Span()): Observable<OpPatch> {
+		switch (params.command) {
+			case 'codeFix':
+				if (!params.arguments || params.arguments.length < 1) {
+					return Observable.throw(new Error(`Command ${params.command} requires arguments`));
+				}
+				return this.executeCodeFixCommand(params.arguments, span);
```

Comment:
Yes, that sounds good! 
I guess you are free to modify codeFix's use of Command arguments at any time without compatibility issues, so there is no need to change anything now.